Sources/MapboxMaps/Style/Style.swift

```diff
@@ -227,22 +249,44 @@ public class Style {
                 as part of the `Result`s success case if the operation is successful.
                 Else, returns a `SourceError` as part of the `Result` failure case.
      */
-    public func getSource<T: Source>(identifier: String, type: T.Type) -> Result<T, SourceError> {
+    public func getSource<T: Source>(identifier: String, type: T.Type = T.self) -> Result<T, SourceError> {
```

Comment:
I'd love to change this function signature to not have to provide a type -- any ideas? One way could be to add some capabilities to the `SourceType` enum to return a type.. And then this method could take in the `SourceType` enum instead (which will provide better type safety).

From an end developer's standpoint, the API would look like:

```swift
mapView.style.getSource(identifier: "some-id", type: .background)
```

Reply:
It should be able to infer the type if the result is assigned to a typed variable

Reply:
Let's iterate.